HW11/src/coffee/order/CoffeeOrderBoard.java

```diff
@@ -0,0 +1,44 @@
+package coffee.order;
+
+import java.util.Iterator;
+import java.util.PriorityQueue;
+
+public class CoffeeOrderBoard {
+
+    private final PriorityQueue<Order> orderPriorityQueue;
+    private int nextOrderNumber;
+
+
+    public CoffeeOrderBoard() {
+        orderPriorityQueue = new PriorityQueue<>();
+        nextOrderNumber = 1;
+    }
+
+    public void add(String name) {
+        Order order = new Order(nextOrderNumber++, name);
+        orderPriorityQueue.offer(order);
+    }
+
+    public Order deliver() {
+        return orderPriorityQueue.poll();
+    }
+
+    public Order deliverByOrderNumber(int number) {
```

Comment:
The method name must be `delivery` (not `deliveryByOrderNumber`) - here, it should be used method overload feature. 

Moreover, in case of method overloading, it is good practice to reuse the method with broader parameters - this approach allows to keep the logic in one place. 

For example: 
```
public Order deliver(int orederNumber){ ... }

public Order deliver(){
  return deliver(nextOrderNumber);
}
```

But this is a flexible rule. For cases when performance matters, it is better to have different implementations.

Reply:
Added fixes according to comments, tested and merged